code/_onclick/click.dm

```diff
@@ -96,7 +96,7 @@
 
 		return
 
-	if(!isturf(loc)) // This is going to stop you from telekinesing from inside a closet, but I don't shed many tears for that
+	if(!isturf(loc) && !find_holder_of_type(A, src)) // Can't touch anything from inside a locker/sleeper etc, unless it's inside our inventory.
```

Comment:
That means you can't attack something on a mob that you are holding.
Any logic that isn't exactly what you want it to be is ultimately fallible: you code the correct system or no system at all.